debezium-server-voyager/debezium-server-voyagerexporter/src/main/java/io/debezium/server/ybexporter/KafkaConnectRecordParser.java

```diff
@@ -128,18 +126,37 @@ public Record parseRecord(Object keyObj, Object valueObj) {
 
     protected void parseEventId(Struct value, Record r) {
         r.eventId = null;
+        // Check if connector is in snapshot mode return
+        if (es.getMode().equals(ExportMode.SNAPSHOT)) {
+            return;
+        }
+
         if (sourceType.equals("oracle") || sourceType.equals("postgresql")) {
             // Extract transaction struct from value if it is available
             Struct transaction = value.getStruct("transaction");
             // Transaction metadata =>
             // Struct{id=0200030002cf0000,total_order=1,data_collection_order=1}
             // Convert id=0200030002cf0000,total_order=1,data_collection_order=1 to string
             if (transaction == null) {
-                return;
+                // Exit the code not the function if transaction struct is not available
+                LOGGER.error("Transaction struct is not available in the event. Exiting.");
```

Comment:
Consider logging the entire json event in readable format.

As a general note: while logging any message always imagine what information an engineer might need to debug an issue if the code-path indeed raises the exception.